python/tk_multi_publish2/api/tree.py

```diff
@@ -338,15 +340,17 @@ class _PublishTreeEncoder(json.JSONEncoder):
     """
     Implements the json encoder interface for custom publish tree serialization.
     """
-    def default(self, publish_tree):
-        return publish_tree.to_dict()
-
-
-class _PublishTreeDecoder(json.JSONDecoder):
-    """
-    Implements the json decoder interface for custom publish tree
-    deserialization.
-    """
-    def decode(self, json_str):
-        tree_dict = sgtk.util.json.loads(json_str)
-        return PublishTree.from_dict(tree_dict)
+    def default(self, data):
+        if isinstance(data, sgtk.Template):
+            return {
+                "__tk_template": True,
```

Comment:
i think it's better to return something along the lines of 

`{_tk_custom_type: template, name: xxxx}`

this way we could extend this in the future to support additional types of objects. 

This does raise the question if the proper solution is to really add support in core to serialize and deserialize templates into json. That solution would clearly be more elegant and would work in a generic way (e.g. not just in the publisher). Did you look into that approach?

Reply:
I didn't, but nothing prevents us from looking into it in the feature. Since this is done as part of the work required to get the webinar working correctly, I recommend we keep the fix local to the app right now. Then we can ticket this up and put it in the grooming pile. When the feature becomes available in core, we can remove the code here and use that.